contrib/raftexample/main.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+	"io/ioutil"
+	"net/http"
+)
+
+// Handler for a http based key-value store backed by raft
+type kvHTTPHandler struct {
+	proposeC	chan<- raftKV
+	sync.RWMutex
+	kvStore		map[string][]byte
```

Comment:
probably we should separate the actual data applier/holder from the http api layer.